File: docs/code-examples/tensor_simple.py
"""Create and log a tensor."""
import numpy as np
import rerun as rr

tensor = np.random.randint(0, 256, (8, 6, 3, 5), dtype=np.uint8)  # 4-dimensional tensor

rr.init("rerun_example_tensor_simple", spawn=True)

# Log the tensor, assigning names to each dimension
rr.log("tensor", rr.Tensor(tensor, dim_names=("width", "height", "channel", "batch")))

File: docs/snippets/all/tensor_simple.rs
//! Create and log a tensor.

use ndarray::{Array, ShapeBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_tensor").spawn()?;

    let mut data = Array::<u8, _>::default((8, 6, 3, 5).f());
    data.map_inplace(|x| *x = rand::random());

    let tensor =
        rerun::Tensor::try_from(data)?.with_dim_names(["width", "height", "channel", "batch"]);
    rec.log("tensor", &tensor)?;

    Ok(())
}
